File: team-500-master/portfolio/src/main/java/nz/ac/canterbury/seng302/portfolio/model/contract/ProjectContract.java
package nz.ac.canterbury.seng302.portfolio.model.contract;

import java.time.Instant;
import java.util.List;

/**
 * A contract for a project.
 *
 * @param id The id of the project.
 * @param name The name of the project.
 * @param description The description of the project.
 * @param startDate The start date of the project.
 * @param endDate The end date of the project.
 * @param sprints The sprints of the project.
 * @param events The events of the project.
 */
public record ProjectContract(
    String id,
    String name,
    String description,
    Instant startDate,
    Instant endDate,
    List<SprintContract> sprints,
    List<EventContract> events,
    List<MilestoneContract> milestones,
    List<DeadlineContract> deadlines
) implements Contractable {}

File: team-500-master/portfolio/src/main/frontend/src/contract/ProjectContract.ts
import {BaseProjectContract} from "./BaseProjectContract";
import {SprintContract} from "./SprintContract";
import {EventContract} from "./EventContract";
import {MilestoneContract} from "./MilestoneContract";
import {DeadlineStore} from "../page/monthly_planner/store/DeadlineStore";
import {DeadlineContract} from "./DeadlineContract";

export interface ProjectContract extends BaseProjectContract {
    id: string
    sprints: SprintContract[]
    events: EventContract[]
    milestones: MilestoneContract[]
    deadlines : DeadlineContract[]
}